chain/wasm/src/transaction/utils.rs

```diff
@@ -68,6 +69,61 @@ impl NativeScript {
     pub fn hash(&self) -> ScriptHash {
         self.0.hash().into()
     }
+
+    pub fn verify(
```

Comment:
I meant just a wrapper around the rust code to avoid duplicating logic e.g.:
```rust
self.0.verify(lower_bound, upper_bound, key_hashes.as_ref())
```

^ untested but it's something like that

We have `AsRef` and `Into`/`From` to convert to/from the rust and WASM structs.